string visitor_id
string building_id
